logdetective/server/database/models/merge_requests.py

```diff
@@ -0,0 +1,462 @@
+        UniqueConstraint(
+            "forge", "project_id", "mr_iid", "job_id", name="uix_mr_project_job"
+        ),
+    )
+
+    comment = relationship(
+        "Comments", back_populates="merge_request", uselist=False
+    )  # 1 comment for 1 job
+
+    @classmethod
+    def create(
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+    ) -> int:
+        """Create a new merge request job entry,
+        returns its PostgreSQL id
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        with transaction(commit=True) as session:
+            mr = cls()
+            mr.forge = forge
+            mr.project_id = project_id
+            mr.mr_iid = mr_iid
+            mr.job_id = job_id
+            session.add(mr)
+            session.flush()
+            return mr.id
+
+    @classmethod
+    def get_by_id(
+        cls,
+        id_: int,
+    ) -> Optional["GitlabMergeRequests"]:
+        """Search for a given PostgreSQL id"""
+        with transaction(commit=False) as session:
+            mr = session.query(cls).filter_by(id=id_).first()
+            return mr
+
+    @classmethod
+    def get_by_details(
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+    ) -> Optional["GitlabMergeRequests"]:
+        """Search for a detailed merge request.
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        with transaction(commit=False) as session:
+            mr = (
+                session.query(cls)
+                .filter_by(
+                    forge=forge, project_id=project_id, mr_iid=mr_iid, job_id=job_id
+                )
+                .first()
+            )
+            return mr
+
+    @classmethod
+    def get_or_create(
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+    ) -> Optional["GitlabMergeRequests"]:
+        """Search for a detailed merge request
+        or create a new one if not found.
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        mr = GitlabMergeRequests.get_by_details(forge, project_id, mr_iid, job_id)
+        if mr is None:
+            id_ = GitlabMergeRequests.create(forge, project_id, mr_iid, job_id)
+            mr = GitlabMergeRequests.get_by_id(id_)
+        return mr
+
+
+class Comments(Base):
+    """Store details for comments
+    created by logdetective in merge requests."""
+
+    __tablename__ = "comments"
+
+    id = Column(Integer, primary_key=True)
+    merge_request_id = Column(
```

Comment:
Same as above; we're really associating with a particular job on a merge request, not the merge request itself.